### PR TITLE
feat: Better error message for flat config plugins

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[x] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Made it so that if the `plugins` key is defined as an array, then we throw a specific validation message that lets people know it's an eslintrc style config that needs to be converted. I also added a section on the config migration guide to include details on using `FlatCompat`.

Refs #17370

#### Is there anything you'd like reviewers to focus on?

Does the wording make sense?

### CODE REVIEW

File: docs/src/use/configure/migration-guide.md [hunk 1]
```diff
@@ -341,6 +341,40 @@ export default [
 ];
 ```
 
+#### Using eslintrc Configs in Flat Config
+
+You may find that there's a shareable config you rely on that hasn't yet been updated to flat config format. In that case, you can use the `FlatCompat` utility to translate the eslintrc format into flat config format. First, install the `@eslint/eslintrc` package:
+
+```shell
+npm install @eslint/js --save-dev
```
Comment (mdjermanovic) [inline docs/src/use/configure/migration-guide.md:349]:
```suggestion
npm install @eslint/eslintrc --save-dev
```

File: messages/eslintrc-plugins.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    return `
+A config object has a "plugins" key defined as an array of strings.
+
+Flat config requires "plugins" to be an object in this form:
+
+    {
+        plugins: {
+            ${plugins[0] || "namespace"}: pluginObject
+        }
+    }
```
Comment (mdjermanovic) [inline messages/eslintrc-plugins.js:14]: Flat config schema throws this error on any array (it doesn't check whether it's an array of _strings_), so if the user specifies plugins like this:

```js
module.exports = [{
    plugins: [somePluginObject]
}];
``` 

The printed message will look like this:

```
Oops! Something went wrong! :(

ESLint: 8.45.0


A config object has a "plugins" key defined as an array of strings.

Flat config requires "plugins" to be an object in this form:

    {
        plugins: {
            [object Object]: pluginObject
        }
    }

Please see the following page for information on how to convert your config object into the correct format:
https://eslint.org/docs/latest/use/configure/migration-guide#importing-plugins-and-custom-parsers

If you're using a shareable config that you cannot rewrite in flat config format, then use the compatibility utility:
https://eslint.org/docs/latest/use/configure/migration-guide#using-eslintrc-configs-in-flat-config
```

Perhaps we should have two templates?

* `plugins` is an array of strings (or empty array?) - possible eslintrc config, use this template
* `plugins` is an array but not array of strings - use another template.

Comment (nzakas) [reply messages/eslintrc-plugins.js]: I think for now I'd like to keep things simple and just assume that when people are using an array then they are most likely using an eslintrc-style config. We can always expand later if we find that's not the case.

I've updated to check if the first item in the array is a string to make sure that the message makes sense.

**Review by aladdin-add — APPROVED** [review]
LGTM.

**Comment by aladdin-add** [comment]
There is a conflict that needs to be resolved.

**Review by mdjermanovic — APPROVED** [review]
LGTM when merge conflicts are resolved.

File: docs/src/use/configure/migration-guide.md [hunk 3]
```diff
@@ -341,6 +341,40 @@ export default [
...
+});
+
+export default [
+
+    // mimic ESLintRC-style extends
+    ...compat.extends("eslint-config-my-config"),
+];
+```
+
+This example uses the `FlatCompat#extends()` method to insert the `eslint-config-my-config` the flat config array.
```
Comment (fasttime) [inline docs/src/use/configure/migration-guide.md:374]:
```suggestion
This example uses the `FlatCompat#extends()` method to insert the `eslint-config-my-config` into the flat config array.
```

**Review by fasttime — COMMENTED** [review]
Just a typo, and a merge conflict that needs to be fixed. Otherwise LGTM.

**Comment by nzakas** [comment]
Should be all set. Please double-check that the merge makes sense.

**Review by fasttime — APPROVED** [review]

**Review by mdjermanovic — APPROVED** [review]

**Pull request merged**